Form toggles: Use admin color schemes' primary color as background

sixhours opened the pull request:
#### Changes proposed in this Pull Request

* Use `var( --color-primary )` as the background color for `@wordpress/components` form toggles so they better match with our color schemes.
* I'm not sure if this is where the override should live, so I'm open to suggestions!

**Before**

<img width="785" alt="Screen Shot 2020-12-18 at 4 38 51 PM" src="https://user-images.githubusercontent.com/2124984/102663818-93091580-414f-11eb-9766-2589002f4949.png">

**After**

<img width="787" alt="Screen Shot 2020-12-18 at 4 37 02 PM" src="https://user-images.githubusercontent.com/2124984/102663676-52110100-414f-11eb-8cc7-2250d4ece851.png">

#### Testing instructions

* Switch to this PR
* Check out toggles across WP.com on simple, Atomic, and Jetpack sites. The Settings area has many of these, but check the Reader or Account contexts, too.
* Change color schemes. The toggles should match the primary color for each color scheme.
* Make sure there are no visual regressions

Fixes #48509

sixhours committed:
Allow wordpress component form toggles to use primary color as background in global styles.

sarayourfriend reviewed (approved):
LGTM! 🚢

sarayourfriend commented:
> I'm not sure if this is where the override should live, so I'm open to suggestions!

I think it's fine where it is but given that these sorts of overrides might grow, it might be good to create a specific `_wp-components-overrides.scss` file for things like this. I'd imagine there are other places we'd want to do these overrides (buttons, for example) and we'd probably not want to continue growing that `_main.scss` file.

sixhours committed:
Move overrides into a separate SCSS file for better organization

sixhours commented:
> I think it's fine where it is but given that these sorts of overrides might grow, it might be good to create a specific _wp-components-overrides.scss file for things like this. I'd imagine there are other places we'd want to do these overrides (buttons, for example) and we'd probably not want to continue growing that _main.scss file.

Good call! I've added that to `style.scss` with a note about what it is and moved the form toggle override there.

sarayourfriend commented on client/assets/stylesheets/style.scss:
```diff
@@ -20,3 +20,6 @@
 @import '~@wordpress/components/build-style/style.css';
+
+// Overrides for @wordpress/components
+@import './_wp-components-overrides.scss';
```
Thanks for doing this!

sarayourfriend reviewed (approved):
LGTM! Thanks for moving it into that new file! 🚀
